### PR TITLE
Add documentation and screenshots

### PR DESCRIPTION
**Pull request by stvstnfrd**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Image Modal XBlock
+==================
+A fullscreen image modal XBlock.
+
+This package is meant to be installed within the OpenEdX platform.
+
+.. image:: ./docs/lms-view-normal.png
```
Comment (stvstnfrd) [inline README.rst]: I don't really like the idea of keeping media assets directly in the repo;
we'll end up w/ a repo >100MB where 99% of the contents are screenshots and screencasts.

Thoughts about alternative storage?
I'd lean toward S3..

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+Installation, system administrator
+----------------------------------
+
+To install the XBlock on your platform,
+add the following to your `requirements.txt` file:
```
Comment (stvstnfrd) [inline README.rst]: Note: I'm intentionally keeping this ambiguous and not going into minutia about `base.in` vs `stanford.txt`, etc..
If you don't know how to install a new package, you're not the target audience here..

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
+Image Modal XBlock
+==================
```
Comment (stvstnfrd) [inline README.rst]: As much as I like markdown, restructured text is a first-class citizen in Python/PyPI land

Trying to use markdown instead means we end up swimming against the current.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+.. image:: ./docs/settings-menu.png
+
+and add the following:
+
+    imagemodal
+
+.. image:: ./docs/advanced-module-list.png
+
+
+Use, Course Staff
```
Comment (stvstnfrd) [inline README.rst]: Right now, I'm not taking time to explain how to add a component to your outline;
is that a fine omission or should it be included?

**Comment by stvstnfrd** [comment]
You can preview the rendered file here:

https://github.com/Stanford-Online/xblock-image-modal/blob/canonize/README.rst

**Comment by stvstnfrd** [comment]
I think it's outside the scope of this PR, but I could imagine filming quick little (<=60s?) screencasts that quickly show how to install the xblock and start using it.
We could just host these videos on youtube and embed them in the `README`.

File: README.rst (hunk 4 above)
Comment (mondiaz) [reply README.rst]: That seems fine to omit to me.

File: README.rst (hunk 1 above)
Comment (caesar2164) [reply README.rst]: Maybe we can switch to using JPGs and also crush them a little? should bring that down to ~250K

**Comment by mondiaz** [comment]
@stvstnfrd I suggest providing distinctions between description and alt text. Instead of 

description (useful for screen readers)
alt text (usef for screen readers)

maybe go with

description (useful for screen readers, longer descriptions)
alt text (useful for screen readers, captions, tags; displays when image does not)

Not sure if it needs spelling out that to use your own images, those should first be uploaded to Files & Uploads.

Also not sure if there are any text limits for the input fields. Description let me enter 10000 words (67186 characters) of Lorem Ipsum text. Seems like that's more than flexible enough FWIW.

File: README.rst (hunk 1 above)
Comment (stvstnfrd) [reply README.rst]: So for v1, I ended up uploading to S3.

As it were, it appears that Github MITMs the URLs, even though they're absolute paths,
and rewrites the paths to go through their CDN. That CDN apparently won't serve images over a certain threshold (~6-7MB?).

So I shrunk the large PNGs by resizing them to to ~1600px (scale preserved).

I still don't have a solution for the GIF;
it's ~25MB

**Comment by stvstnfrd** [comment]
@mondiaz Thanks! I went with your text for description/alt_text.

Since this is only intended for an at least somewhat technical audience,
I'm fine leaving URL "ambiguous" for now, since like in the example,
the file _may_ come from elsewhere.

Re: text limits
We (the team at Stanford) don't impose any specific restrictions on string length there.
Whatever limits there may be, we "inherit" from edx's default behavior.
If it's generous enough and we're not explicitly handling it,
I'd prefer to leave it unmentioned for now :)

**Comment by stvstnfrd** [comment]
@mondiaz @caesar2164 I have also added:

- a CMS demo video
- an LMS demo video
- howto add component to studio (cribbed from demo course)
- a better summary (cribbed from demo course)

**Comment by mondiaz** [comment]
@stvstnfrd the added vids + text look good to me, thanks! 👍

**Pull request merged**